subprojects/packagefiles/c-ares/meson.build

```diff
@@ -0,0 +1,230 @@
+  'src/lib/ares__buf.c',
+  'src/lib/ares__readaddrinfo.c',
+  'src/lib/ares__slist.c',
+  'src/lib/ares__sortaddrinfo.c',
+  'src/lib/ares__read_line.c',
+  'src/lib/ares__timeval.c',
+  'src/lib/ares_android.c',
+  'src/lib/ares_cancel.c',
+  'src/lib/ares_data.c',
+  'src/lib/ares_destroy.c',
+  'src/lib/ares_expand_name.c',
+  'src/lib/ares_expand_string.c',
+  'src/lib/ares_fds.c',
+  'src/lib/ares_free_hostent.c',
+  'src/lib/ares_free_string.c',
+  'src/lib/ares_freeaddrinfo.c',
+  'src/lib/ares_getaddrinfo.c',
+  'src/lib/ares_getenv.c',
+  'src/lib/ares_gethostbyaddr.c',
+  'src/lib/ares_gethostbyname.c',
+  'src/lib/ares_getnameinfo.c',
+  'src/lib/ares_getsock.c',
+  'src/lib/ares_init.c',
+  'src/lib/ares_library_init.c',
+  'src/lib/ares_mkquery.c',
+  'src/lib/ares_create_query.c',
+  'src/lib/ares_nowarn.c',
+  'src/lib/ares_options.c',
+  'src/lib/ares_parse_a_reply.c',
+  'src/lib/ares_parse_aaaa_reply.c',
+  'src/lib/ares_parse_caa_reply.c',
+  'src/lib/ares_parse_mx_reply.c',
+  'src/lib/ares_parse_naptr_reply.c',
+  'src/lib/ares_parse_ns_reply.c',
+  'src/lib/ares_parse_ptr_reply.c',
+  'src/lib/ares_parse_soa_reply.c',
+  'src/lib/ares_parse_srv_reply.c',
+  'src/lib/ares_parse_txt_reply.c',
+  'src/lib/ares_parse_uri_reply.c',
+  'src/lib/ares_platform.c',
+  'src/lib/ares_process.c',
+  'src/lib/ares_query.c',
+  'src/lib/ares_rand.c',
+  'src/lib/ares_search.c',
+  'src/lib/ares_send.c',
+  'src/lib/ares_strcasecmp.c',
+  'src/lib/ares_strdup.c',
+  'src/lib/ares_strerror.c',
+  'src/lib/ares_strsplit.c',
+  'src/lib/ares_timeout.c',
+  'src/lib/ares_version.c',
+  'src/lib/bitncmp.c',
+  'src/lib/inet_net_pton.c',
+  'src/lib/inet_ntop.c',
+  'src/lib/windows_port.c')
+
+cares_headers = files(
+  'src/lib/ares__htable.h',
+  'src/lib/ares__htable_asvp.h',
+  'src/lib/ares__htable_stvp.h',
+  'src/lib/ares__llist.h',
+  'src/lib/ares__buf.h',
+  'src/lib/ares__slist.h',
+  'src/lib/ares_android.h',
+  'src/lib/ares_data.h',
+  'src/lib/ares_getenv.h',
+  'src/lib/ares_inet_net_pton.h',
+  'src/lib/ares_iphlpapi.h',
+  'src/lib/ares_ipv6.h',
+  'src/lib/ares_nowarn.h',
+  'src/lib/ares_platform.h',
+  'src/lib/ares_private.h',
+  'src/lib/ares_strcasecmp.h',
+  'src/lib/ares_strdup.h',
+  'src/lib/ares_strsplit.h',
+  'src/lib/bitncmp.h',
+  'src/lib/ares_setup.h',
+  'src/lib/setup_once.h')
+
+cares_include_dirs = include_directories('include', 'src/lib')
+
+cc = meson.get_compiler('c')
+cares_args_map = {
+'common': [
+  '-DCARES_BUILDING_LIBRARY',
+
+  cc.has_header('arpa/inet.h') ? '-DHAVE_ARPA_INET_H' : [],
+  cc.has_header('errno.h') ? '-DHAVE_ERRNO_H' : [],
+  cc.has_header('limits.h') ? '-DHAVE_LIMITS_H' : [],
+  cc.has_header('netdb.h') ? '-DHAVE_NETDB_H' : [],
+  cc.has_header('netinet/in.h') ? '-DHAVE_NETINET_IN_H' : [],
+  cc.has_header('netinet/tcp.h') ? '-DHAVE_NETINET_TCP_H' : [],
+  cc.has_header('stdint.h') ? '-DHAVE_STDINT_H' : [],
+  cc.has_header('strings.h') ? '-DHAVE_STRINGS_H' : [],
```

Comment:
why are these not in a loop?

Reply:
Because loop does no allow to do it in-place, and will not make it look simpler